WebKitBrowser/HtmlApp.config

```diff
@@ -33,12 +33,12 @@ map()
     kv(xhrcache ${PLUGIN_WEBKITBROWSER_XHRCACHE})
     kv(webgl ${PLUGIN_WEBKITBROWSER_WEBGL})
     kv(threadedpainting ${PLUGIN_WEBKITBROWSER_THREADEDPAINTING})
-    if (PLUGIN_WEBKITBROWSER_DISKCACHEDIR)
-      kv(diskcachedir ${PLUGIN_WEBKITBROWSER_DISKCACHEDIR})
```

Comment:
Think this was a redundant setting, it was also triggered somewhere else, let revisit this.

Reply:
In our case we need to configure disc cache in a different location(persistent storage is rather small). We do need this.

Reply:
We looked a little deeper into this, for this kind of functionality we added the plugin "persistentpathpostfix" and "volatilepathpostfix" options where for a plugin you can overrule the path being extended with another name instead of the callsign (which is the default), and therefore enable that two plugins (so also WebKit tabs) can share the same persistant (and volatile path). 
Would you then still like to be able to beneath this now already shared path add another configurable name for the DiskCache? And as the cache folder is set by the XDG_CACHE_HOME env variable is that standard functionality in WebKit (or for example added with a patch)?

Reply:
we need to set disk cache dir to point to the SD card location (is not volatile), while the persistent path points to the internal flash. setting XDG_CACHE_HOME is also important so the GStreamer registry is stored in disk cache location. If you don't need it don't use it :)

Reply:
Okay understand. As we do not want people to use the configuration to go outside the sandbox as defined by Thunder I'll add the setting to the default configuration and it will add up the standard disk cache dir. You can then use this unique path to setup a symbolic link to the desired location outside the sandbox.

Reply:
> As we do not want people to use the configuration to go outside the sandbox as defined by Thunder

the only way to do it is to enforce it through the container... 

> I'll add the setting to the default configuration and it will add up the standard disk cache dir

It sounds like a hack to escape from the configuration. I think it is better to have it explicitly configured. Don't configure it if you don't want, just keep it as it is. We already use this config.

Reply:
In the Thunder architecture by default all plugins write volatile and persistent paths that can be configured as a whole. Everywhere where you are able to override part of this path it is withing the section assigned to that plugin. If we would break that here that would make the architecture inconstent and harder to understand, you would not expect this single setting to go outside the sandbox. As mentioned if that is however explicitly required as an exception there is an easy way to do that with a symbolic link. The setting for the DiskCache dir then enables you to pick this up as the unique source for the symbolic link. But was this feature discussed with Pierre before it was added?

Reply:
This is about a use case that is not covered by the Thunder default configuration. What you suggest is a hacky workaround in my opinion. And yes, this was reviewed and approved by Metro as well - https://github.com/rdkcentral/rdkservices/pull/186

Reply:
This was approved by metro as indeed in the past this was not possible. It was always in the sandboxed environment. Hence why I asked you to raise an issue so we could improve on this valid use case in the future (Ticket: https://github.com/rdkcentral/Thunder/issues/154) and I do think this is a valid use case that we need to tackle, so we did.
Thunder has several focus point but one related to this is uniformity. To make sure that *all* plugins behave uniform we create an environment for the plugin. All plugins obey the rules:
Persistent -> PersitentPath() /* data that remains over reboots, typically nand flash/HDD  */
Volatile -> VolatilePath() /* data remains as long as the box is on, typically memory */
Data -> DataPath() /* Read only can not be changed, added during the creation of the image, typically ROMSquash */

If you build a plugin you adhere to these rules, we do not want the plugin builder to wander over the whole system (security). Depending on his needs, he can select one of the above categories and he should prefix the path with the respective IShell method (PeritentPath()/VolatilePath()/DataPath())
We did that properly for the WebKit plugin as well, till you came up with this valid use case. At that time it was not possible so you raised the issue, shown above)
We tackled it by adding new config variables to the base of a plugin (so not part of the WebKit config, it is a generic feature, can be use by any Thunder plugin) at the base we can now define persistentpostfix / volatilepostfix as part of the generic plugin configuration (see https://github.com/rdkcentral/Thunder/pull/229). Using this feature you can configure your SearchAndDiscoveryApp and ResidentApp and any other config that needs to share persistent information (or volatile) information with the "persistentpostfix:" "ComcastSharedCache" on all the plugins. Than the PersitentPath() offered by the system will still use the system wide configured PersistentPath but adds the postfix, i.s.o. the callsign effectivly sharing the storage between the plugins.
This again is in line with the Thunder philosophy where you only have to configure once and enjoy multiple deployments without re-iterating where your persistent location is and still have the possibility to share persistent or volatile storage between plugins.
So to meet your use case, the only think you have to do is add the following lines to the plugin config that need to share Volatile or Persitent data:
"persistentpostfix": "ComcastSharedCache" and/or "volatilepostfix": "ComcastSharedCache" and all plugins with these keywords will be directed to <persitentPath, from the config.json, system wide>/<classname of the plugin>/ComcastSharedCache/  (classname is consitent for all your tabs, it is not the callsign, if you where wondering :-)

Reply:
@pwielders, if  my understanding correct:

Persistent - path for local storage, cookie db etc, read/write (secure location in our case, rather small)
Volatile - basically tmpfs (/tmp)
Data - read-only, location for injected bundle/fonts/etc. /usr/share/...

We need one more to configure DataCache dir (what #186 adds). where WebKit can store cached images/js/css files. This can be persistent across the reboots if the device supports it, the location is different from local storage/db, etc. It is not a tmpfs.